benchmark/rocblas/blas3/trsm.cpp

```diff
@@ -0,0 +1,211 @@
+static inline void rocblas_trsm_f(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CHECK_ROCBLAS_STATUS(rocblas_strsm(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CHECK_ROCBLAS_STATUS(rocblas_dtrsm(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, rocblas_handle& rb_handle, char side,
+         char uplo, char trans, char diag, index_t m, index_t n, scalar_t alpha,
+         bool* success) {
+  // Standard test setup.
+  index_t lda = (side == 'l') ? m : n;
+  index_t ldb = m;
+  index_t k = (side == 'l') ? m : n;
+
+  {
+    // The counters are double. We convert m, n and k to double to avoid
+    // integer overflows for n_fl_ops and bytes_processed
+    const double m_d = static_cast<double>(m);
+    const double n_d = static_cast<double>(n);
+    const double k_d = static_cast<double>(k);
+
+    state.counters["m"] = m_d;
+    state.counters["n"] = n_d;
+
+    const double mem_readA = k_d * (k_d + 1) / 2;
+    const double mem_readWriteB = 2 * m_d * n_d;
+    const double total_mem = (mem_readA * mem_readWriteB) * sizeof(scalar_t);
+    state.counters["bytes_processed"] = total_mem;
+    state.SetBytesProcessed(state.iterations() * total_mem);
+
+    const double nflops_AtimesB =
+        2 * k_d * (k_d + 1) / 2 * (side == 'l' ? n_d : m_d);
+    const double nflops_timesAlpha = m_d * n_d;
+    const double total_nflops = nflops_AtimesB + nflops_timesAlpha;
+    state.counters["n_fl_ops"] = total_nflops;
+    state.SetItemsProcessed(state.iterations() * total_nflops);
+  }
+
+  // Matrix options (rocBLAS)
+  const rocblas_side side_rb =
+      (side == 'l') ? rocblas_side_left : rocblas_side_right;
+  const rocblas_fill uplo_rb =
+      (uplo == 'u') ? rocblas_fill_upper : rocblas_fill_lower;
+  const rocblas_operation trans_rb =
+      (trans == 'n') ? rocblas_operation_none : rocblas_operation_transpose;
+  const rocblas_diagonal diag_rb =
+      (diag == 'u') ? rocblas_diagonal_unit : rocblas_diagonal_non_unit;
+
+  // Data sizes
+  const int a_size = k * lda;
+  const int b_size = ldb * n;
+
+  // Matrices
+  std::vector<scalar_t> a(a_size);
+  std::vector<scalar_t> b =
+      blas_benchmark::utils::random_data<scalar_t>(b_size);
+
+  // Populate the main input diagonal.
+  const scalar_t diag_value =
+      diag == 'u' ? scalar_t{1}
+                  : blas_benchmark::utils::random_scalar<scalar_t>(
+                        scalar_t{1}, scalar_t{10});
+
+  blas_benchmark::utils::fill_trsm_matrix(a, k, lda, uplo, diag_value,
+                                          scalar_t{0});
+
+  {
+    // Device memory allocation & H2D copy
+    blas_benchmark::utils::HIPVector<scalar_t> a_gpu(a_size, a.data());
+    blas_benchmark::utils::HIPVector<scalar_t> b_gpu(b_size, b.data());
+
+    CHECK_ROCBLAS_STATUS(
+        rocblas_set_pointer_mode(rb_handle, rocblas_pointer_mode_host));
```

Comment:
is this necessary?

Reply:
It looks like it's not. Couldn't find in the doc if it's set to `rocblas_pointer_mode_host`by default but results show no difference between enabling/disabling it. Will remove it then thanks!